python/cuml/linear_model/logistic_regression.pyx

```diff
@@ -314,6 +337,38 @@ class LogisticRegression(Base):
         """
         return self.qn.predict(X, convert_dtype=convert_dtype)
 
+    def predict_proba(self, X, convert_dtype=False):
+        """
+        Predicts the class probabilities for X
+
+        Parameters
+        ----------
+        X : array-like (device or host) shape = (n_samples, n_features)
+            Dense matrix (floats or doubles) of shape (n_samples, n_features).
+            Acceptable formats: cuDF DataFrame, NumPy ndarray, Numba device
+            ndarray, cuda array interface compliant array like CuPy
+
+        convert_dtype : bool, optional (default = False)
+            When set to True, the predict method will, when necessary, convert
+            the input to the data type which was used to train the model. This
+            will increase memory used for the method.
+        Returns
+        ----------
+        y: cuDF DataFrame
+           Dense matrix (floats or doubles) of shape (n_samples, C)
+        """
+        proba = self.decision_function(X, convert_dtype=convert_dtype)
+        proba = cp.asarray(proba.to_gpu_matrix())
```

Comment:
we should probably just return gpu matrices for now... this is a pain that you have to keep converting back and forth. If both decision_function and predict_proba returned gpu matrices, I think that would simplify the code, right?

Reply:
Divye & I were just discussing this. I was suggesting using the `_prep_output` function from UMAP to make the return match the input until the Array is merged but I like your idea better.